src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -142,47 +165,75 @@ function run_coverage() {
-  # http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php
-  # The expected result can be constructed manually by following the lcov
-  # documentation and manually checking what lines of code are covered when
-  # running the test.
-  cat <<EOF > expected_result.dat
-TN:
-SF:coverage_srcs/a.cc
-FN:3,_Z1ab
-FNDA:1,_Z1ab
-FNF:1
-FNH:1
-DA:3,1
-DA:4,1
-DA:5,1
-DA:7,0
-LF:4
-LH:3
-end_of_record
-TN:
-SF:coverage_srcs/t.cc
-FN:4,main
-FNDA:1,main
-FNF:1
-FNH:1
-DA:4,1
-DA:5,1
-DA:6,1
-LF:3
-LH:3
-end_of_record
-EOF
-
-  # tools/test/collect_cc_coverage.sh places the coverage result in
-  # $COVERAGE_OUTPUT_FILE
-  diff -u expected_result.dat "$COVERAGE_OUTPUT_FILE_VAR" >> "$TEST_log" \
-    || fail "Coverage output file is different than the expected file"
+    # Run the C++ coverage script with the environment setup accordingly.
+    # This will get coverage results for coverage_srcs/a.cc and
+    # coverage_srcs/t.cc.
+    run_coverage > "$TEST_log"
+
+    # Assert that the C++ coverage output is correct.
+    # The covered files are coverage_srcs/a.cc and coverage_srcs/t.cc.
+
+   # The expected total number of lines of COVERAGE_OUTPUT_FILE
+   # is 25. This number can be computed by counting the number
+   # of lines in the variables declared below $expected_result_a_cc
+   # and $expected_result_t_cc.
+   [[ $(wc -l < "$COVERAGE_OUTPUT_FILE_VAR") == 25 ]] || \
+      fail "Number of lines $nr_of_lines is different than 25"
+
+    # The expected result can be constructed manually by following the lcov
+    # documentation and manually checking what lines of code are covered when
+    # running the test.
+    # For details about the lcov format see
+    # http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php
+    # The newlines are replaced with commas to facilitate comparing the
+    # expected values with the actual results.
+
+    # The expected coverage result for coverage_srcs/a.cc.
+    local expected_result_a_cc="TN:,\
+SF:coverage_srcs/a.cc,\
+FN:3,_Z1ab,\
+FNDA:1,_Z1ab,\
+FNF:1,\
+FNH:1,\
+DA:3,1,\
+DA:4,1,\
+DA:5,1,\
+DA:7,0,\
+LF:4,\
+LH:3,\
+end_of_record"
+
+    # The expected coverage result for coverage_srcs/t.cc.
+    local expected_result_t_cc="TN:,\
+SF:coverage_srcs/t.cc,\
+FN:4,main,\
+FNDA:1,main,\
+FNF:1,\
+FNH:1,\
+DA:4,1,\
+DA:5,1,\
+DA:6,1,\
+LF:3,\
+LH:3,\
+end_of_record"
+
+    # Assert that the coverage output file contains the coverage data for the
+    # two cc files: coverage_srcs/a.cc and coverage_srcs/t.cc.
+    # The C++ coverage script places the final coverage result in
+    # $COVERAGE_OUTPUT_FILE.
+    #
+    # The result for each source file must be asserted separately because the
+    # coverage tools (e.g. lcov, gcov) do not guarantee any particular order.
+    # The order can differ for example based on OS or version. The source files
+    # order in the coverage report is not relevant.
+    #
+    # tr '\n' , < "$COVERAGE_OUTPUT_FILE_VAR" replaces each newline of the coverage
```

Comment:
Remove this comment, that's what the 'man' pages are for, there's nothing non-trivial happening here.